app/src/main/java/com/github/se/eduverse/ui/todo/TodoListScreen.kt

```diff
@@ -160,3 +198,91 @@ fun AddTodoEntry(onAdd: (String) -> Unit) {
         }
       }
 }
+
+/**
+ * Composable that represents the options menu to display when the TodoItem options button is
+ * clicked
+ *
+ * @param expanded whether the menu is expanded
+ * @param onDismiss code executed when the menu is dismissed
+ * @param onDelete code executed when the delete option is clicked
+ * @param onRename code executed when the rename option is clicked
+ */
+@Composable
+fun TodoOptionsMenu(
+    todoId: String,
+    expanded: Boolean,
+    onDismiss: () -> Unit,
+    onDelete: () -> Unit,
+    onRename: () -> Unit,
+) {
+  DropdownMenu(
+      expanded = expanded,
+      onDismissRequest = { onDismiss() },
+      properties = PopupProperties(focusable = false)) {
+        DropdownMenuItem(
+            text = { Text("Delete", color = Color.Red) },
+            onClick = {
+              onDelete()
+              onDismiss()
+            },
+            modifier = Modifier.testTag("deleteTodoButton_${todoId}"))
+
+        DropdownMenuItem(
+            text = { Text("Rename") },
+            onClick = {
+              onRename()
+              onDismiss()
+            },
+            modifier = Modifier.testTag("renameTodoButton_${todoId}"))
+      }
+}
+
+/**
+ * Composable that represents a dialog to rename a todo
+ *
+ * @param todoName the current name of the todo
+ * @param onRename code executed when the rename button is clicked
+ * @param onDismiss code executed when the dialog is dismissed
+ */
+@Composable
+fun RenameTodoDialog(todoName: String, onRename: (String) -> Unit, onDismiss: () -> Unit) {
+  var newName by remember { mutableStateOf(TextFieldValue(todoName, TextRange(todoName.length))) }
```

Comment:
I feel like it is better to hold data inside a state flow, but if it works it's ok

Reply:
TextFieldValue is not a function, it's a class that holds information about the value in the text field, so here the mutable state flow holds an instance of the class